### PR TITLE
Increase codecov coverage

### PR DESCRIPTION
**Pull request by lillian542**
**Context:**
After updating to the new device API, codecov was unhappy with the lost coverage [here](https://app.codecov.io/gh/PennyLaneAI/pennylane/pull/4436/indirect-changes?src=pr&el=tree-more). Coverage loss was reported in:

1. testing TF support for `compute_eigvals` for `PCPhase` and `IsingZZ`
2. one line in interfaces/torch.py
3. one line in transforms/hamiltonian_expand.py

**Description of the Change:**

Added tests for points 1 and 3. 

Reintroduced running `test_vqe.py` test `test_optimize_torch` with DQL (in addition to DQ2) to maintain code coverage from before the device swap, as its not clear if that line is reachable with DQ2.

**Benefits:**

Code coverage didn't decrease when switching device API

### CODE REVIEW

**Comment by lillian542** [comment]
[sc-45902]

File: tests/transforms/test_hamiltonian_expand.py [hunk 1]
```diff
@@ -292,6 +292,22 @@ def test_hamiltonian_dif_tensorflow(self):
...
+        H = qml.Hamiltonian([1], [qml.PauliX(0)])
+        tape = qml.tape.QuantumTape([], [qml.expval(H)])
+
+        tape, processing_fn = qml.transforms.hamiltonian_expand(tape)
+
+        # get dimensions of execution result
+        r = len(dev.execute(tape))
+
+        # create a "result" whose dimensions force the condition len(r_group) != 1)
+        processing_fn([np.ones(r + 1)])
```
Comment (mudit2812) [inline tests/transforms/test_hamiltonian_expand.py:309]: Is anything supposed to be validated after this?

**Review by mudit2812 — COMMENTED** [review]
Just one question. Otherwise looks good. Happy to approve once the comment is addressed.

File: tests/transforms/test_hamiltonian_expand.py (hunk 1 above)
Comment (lillian542) [reply tests/transforms/test_hamiltonian_expand.py:309]: I guess I was mostly making sure it runs without raising an error, but I can add an assertion after with what the output should be. This test is very forced, tbh. I don't love it.

File: tests/transforms/test_hamiltonian_expand.py
Comment (albi3ro) [inline tests/transforms/test_hamiltonian_expand.py:1]: Recommended test case:

```
dev = qml.device("default.qubit", shots=shots)

H = qml.Hamiltonian([1, 2.0], [qml.PauliZ(0), qml.PauliX(0)])
H.compute_grouping()

@qml.transforms.hamiltonian_expand
@qml.qnode(dev)
def circuit():
    return qml.expval(H)

circuit()
```

Comment (lillian542) [reply tests/transforms/test_hamiltonian_expand.py:1]: Thanks, that's much better :) Updated!

File: tests/transforms/test_hamiltonian_expand.py (hunk 1 above)
Comment (lillian542) [reply tests/transforms/test_hamiltonian_expand.py]: Replaced with a better suggestion from Christina below

**Review by mudit2812 — APPROVED** [review]

File: tests/ops/qubit/test_parametric_ops.py [hunk 2]
```diff
@@ -1247,6 +1247,23 @@ def test_pswap_eigvals_jax(self, phi):
         evs_expected = [1, 1, -qml.math.exp(1j * phi), qml.math.exp(1j * phi)]
         assert qml.math.allclose(evs, evs_expected)
 
+    @pytest.mark.tf
+    @pytest.mark.parametrize("phi", np.linspace(-np.pi, np.pi, 10))
+    def test_pcphase_eigvals_tf(self, phi):
+        """Test eigenvalues computation for PCPhase using Tensorflow interface"""
+        import tensorflow as tf
+
+        phi = 0.5432
```
Comment (timmysilv) [inline tests/ops/qubit/test_parametric_ops.py:1256]: defined in test parametrization already
```suggestion
```

**Review by timmysilv — COMMENTED** [review]
i'm not saying you should definitely remove that one line in `torch.py`, but if you remove it and CI doesn't fail... maybe it doesn't need to be there? Probably don't remove it. I would wager a good amount that someone will hit a bug in a year because of it and not remember this PR, but idk. It bothers me : (

**Comment by albi3ro** [comment]
> i'm not saying you should definitely remove that one line in `torch.py`, but if you remove it and CI doesn't fail... maybe it doesn't need to be there? Probably don't remove it. I would wager a good amount that someone will hit a bug in a year because of it and not remember this PR, but idk. It bothers me : (

I figured out the hamiltonian thing by checking out the last release, raising a warning on the line, and then running the test suite. Might be an option here.

I am rewriting the the torch interface in #4654 , so it might be nice to know if I'm actually covering all the cases I'm needing to cover.

**Review by timmysilv — APPROVED** [review]

**Review by albi3ro — APPROVED** [review]

**Pull request merged**